samples/openvx/wrappers_video.cpp

```diff
@@ -0,0 +1,251 @@
+    //Node::create(graph, VX_KERNEL_GAUSSIAN_3x3, inputImage, gb);
+    Node::create(graph, VX_KERNEL_SOBEL_3x3, gb, gx, gy);
+    Node::create(graph, VX_KERNEL_MAGNITUDE, gx, gy, mag);
+    Node::create(graph, VX_KERNEL_PHASE, gx, gy, ang);
+    Node::create(graph, VX_KERNEL_CONVERTDEPTH, mag, mag8,
+                 ivx::Scalar::create<VX_TYPE_ENUM>(context, VX_CONVERT_POLICY_SATURATE),
+                 shift);
+    Node::create(graph, VX_KERNEL_THRESHOLD, ang, threshAng, thra);
+    Node::create(graph, VX_KERNEL_THRESHOLD, mag8, threshMag, thrm);
+    Node::create(graph, VX_KERNEL_AND, thra, thrm, thrv);
+    Node::create(graph, VX_KERNEL_AND, gray, thrv, outputImage);
+
+    graph.verify();
+
+    return graph;
+}
+
+
+int ovxDemo(std::string inputPath, bool useCopy, bool mapFromVx)
+{
+    using namespace cv;
+    using namespace ivx;
+
+    Mat frame;
+    VideoCapture vc(inputPath);
+    if (!vc.isOpened())
+        return -1;
+
+    vc >> frame;
+    if (frame.empty()) return -1;
+
+    //check frame format
+    if (frame.type() != CV_8UC3) return -1;
+
+    try
+    {
+        Context context = Context::create();
+        //put user data from cv::Mat to vx_image
+        vx_df_image color = Image::matTypeToFormat(frame.type());
+        vx_uint32 width = frame.cols, height = frame.rows;
+        Image ivxImage;
+        if (useCopy)
+        {
+            ivxImage = Image::create(context, width, height, color);
+        }
+        else
+        {
+            vx_imagepatch_addressing_t addressing = Image::createAddressing(width, height, (vx_uint32)frame.elemSize(),
+                                                                            (vx_uint32)frame.step);
+            const std::vector<vx_imagepatch_addressing_t> addrs(1, addressing);
+            const std::vector<void*> ptrs(1, frame.data);
+            ivxImage = Image::createFromHandle(context, color, addrs, ptrs);
+        }
+
+        Image ivxResult;
+
+        Mat output;
+        if (useCopy || mapFromVx)
+        {
+            //we will copy or map data from vx_image to cv::Mat
+            ivxResult = ivx::Image::create(context, width, height, VX_DF_IMAGE_U8);
+        }
+        else if (!mapFromVx)
+        {
+            //create vx_image based on user data, no copying required
+            output = cv::Mat(height, width, CV_8U, cv::Scalar(0));
+            vx_imagepatch_addressing_t addressing = Image::createAddressing(width, height, (vx_uint32)output.elemSize(),
+                                                                            (vx_uint32)output.step);
+            const std::vector<vx_imagepatch_addressing_t> addrs(1, addressing);
+            const std::vector<void*> ptrs(1, output.data);
+            ivxResult = Image::createFromHandle(context, Image::matTypeToFormat(CV_8U), addrs, ptrs);
+        }
+
+        Graph graph = createProcessingGraph(context, ivxImage, ivxResult);
+
+        std::vector<void*> ptrs(ivxResult.planes(), (void*)0xbad);
```

Comment:
According to OpenVX standard, the argument `newPtrs` of `vxSwapImageHandle` should be NULL or point to several non-NULL pointers (even if it does not use them).
But we cannot pass the arguments `newPtrs == NULL` and `prevPtrs != NULL` at the same time because this is forbidden in current wrappers.
@apavlenko should I add such possibility to wrappers in this PR?